File: Android/app/src/main/java/ru/inwords/inwords/domain/interactor/translation/TranslationWordsInteractor.java
package ru.inwords.inwords.domain.interactor.translation;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Observable;
import ru.inwords.inwords.data.dto.WordTranslation;

public interface TranslationWordsInteractor {
    Completable addReplace(WordTranslation wordTranslation);

    Completable remove(WordTranslation wordTranslation);

    Completable update(WordTranslation oldWord, WordTranslation newWord);

    Observable<List<WordTranslation>> getAllWords();
}

File: Android/app/src/main/java/ru/inwords/inwords/domain/interactor/translation/TranslationWordsInteractor.kt
package ru.inwords.inwords.domain.interactor.translation

import io.reactivex.Completable
import io.reactivex.Observable
import ru.inwords.inwords.data.dto.WordTranslation

interface TranslationWordsInteractor {
    fun getAllWords(): Observable<List<WordTranslation>>

    fun addReplace(wordTranslation: WordTranslation): Completable

    fun addReplaceAll(wordTranslations: List<WordTranslation>): Completable

    fun remove(wordTranslation: WordTranslation): Completable

    fun removeAll(wordTranslations: List<WordTranslation>): Completable

    fun update(oldWord: WordTranslation, newWord: WordTranslation): Completable

    fun clearCache()
}
